vault/identity_store_oidc.go

```diff
@@ -556,6 +575,34 @@ func (i *IdentityStore) pathOIDCReadKey(ctx context.Context, req *logical.Reques
 	}, nil
 }
 
+// getRolesReferencingTargetKeyName returns a slice of roles referenced by targetKeyName.
+// Note: this is not threadsafe. It is to be called with Lock already held.
+func (i *IdentityStore) getRolesReferencingTargetKeyName(ctx context.Context, req *logical.Request, targetKeyName string) ([]role, error) {
+	roleNames, err := req.Storage.List(ctx, roleConfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var tempRole role
+	rolesReferencingTargetKeyName := make([]role, 0)
```

Comment:
Nit: since you're appending here, we don't need to pre-alloc the slice (i.e. `var rolesReferencingTargetKeyName []role` should work just fine).

Reply:
good call!